broker/lib/controllers/ServiceFabrikApiController.js

```diff
@@ -249,45 +347,210 @@ class ServiceFabrikApiController extends FabrikBaseController {
+      })
+      .catchThrow(new NotFound(`No backup found for service instance '${req.params.instance_id}'`));
+  }
+
+  getLastBackupV1(req, res) {
     req.manager.verifyFeatureSupport('backup');
-    const instance_id = req.params.instance_id;
+    const instanceId = req.params.instance_id;
     const noCache = req.query.no_cache === 'true' ? true : false;
-    const tenant_id = req.entity.tenant_id;
+    const tenantId = req.entity.tenant_id;
     return req.manager
-      .getLastBackup(tenant_id, instance_id, noCache)
+      .getLastBackup(tenantId, instanceId, noCache)
       .then(result => res
-        .status(200)
+        .status(CONST.HTTP_STATUS_CODE.OK)
         .send(_.omit(result, 'secret', 'agent_ip'))
       )
-      .catchThrow(NotFound, new NotFound(`No backup found for service instance '${instance_id}'`));
+      .catchThrow(NotFound, new NotFound(`No backup found for service instance '${instanceId}'`));
   }
 
   abortLastBackup(req, res) {
+    if (config.enable_service_fabrik_v2) {
+      return this.abortLastBackupV2(req, res);
+    }
+    return this.abortLastBackupV1(req, res);
+  }
+
+  abortLastBackupV1(req, res) {
     req.manager.verifyFeatureSupport('backup');
-    const instance_id = req.params.instance_id;
-    const tenant_id = req.entity.tenant_id;
+    const instanceId = req.params.instance_id;
+    const tenantId = req.entity.tenant_id;
     return req.manager
-      .abortLastBackup(tenant_id, instance_id)
+      .abortLastBackup(tenantId, instanceId)
       .then(result => res
-        .status(result.state === 'aborting' ? 202 : 200)
+        .status(result.state === 'aborting' ? CONST.HTTP_STATUS_CODE.ACCEPTED : CONST.HTTP_STATUS_CODE.OK)
         .send({})
       );
   }
 
+
+  abortLastBackupV2(req, res) {
+    req.manager.verifyFeatureSupport('backup');
+    const backupStartedAt = new Date();
+    return eventmesh.apiServerClient.getLastOperation({
+      resourceId: req.params.instance_id,
+      operationName: CONST.OPERATION_TYPE.BACKUP,
+      operationType: CONST.APISERVER.RESOURCE_TYPES.DEFAULT_BACKUP,
+    }).then(backupGuid => {
+      return eventmesh.apiServerClient.getOperationState({
+        operationName: CONST.OPERATION_TYPE.BACKUP,
+        operationType: CONST.APISERVER.RESOURCE_TYPES.DEFAULT_BACKUP,
+        operationId: backupGuid,
+      }).then(state => {
+        // abort only if the state is in progress
+        if (state === CONST.APISERVER.RESOURCE_STATE.IN_PROGRESS) {
+          return eventmesh.apiServerClient.updateOperationState({
+            operationName: CONST.OPERATION_TYPE.BACKUP,
+            operationType: CONST.APISERVER.RESOURCE_TYPES.DEFAULT_BACKUP,
+            operationId: backupGuid,
+            stateValue: CONST.OPERATION.ABORT
+          });
+        } else {
+          logger.info(`Skipping abort as state is : ${state}`);
+        }
+      }).then(() => ServiceFabrikApiController.getResourceOperationStatus({
```

Comment:
can get rid of it and if we have save the state in api server then return 'ACCEPTED' HTTP Status code. I am not sure what difference 200 vs 202 is going do. You send 200 in case request abort is not initated?I know this is inline with exisitng implementation but not sure how the invoker is to interpret 200 vs 202.